app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -93,11 +93,18 @@ def show
     param :id, :identifier, :desc => N_("content view numeric identifier"), :required => true
     param :name, String, :desc => N_("module name to restrict modules for"), :required => false
     def available_puppet_modules
-      current_uuids = @view.content_view_puppet_modules.where("uuid is NOT NULL").pluck(:uuid)
+      current_cv_puppet_modules = @view.content_view_puppet_modules.where("uuid is NOT NULL")
+      current_uuids = current_cv_puppet_modules.pluck(:uuid)
       repositories = @view.organization.library.puppet_repositories
 
       query = PuppetModule.in_repositories(repositories)
-      query = query.where(:name => params[:name]) if params[:name]
+      if params[:name]
+        query = query.where(:name => params[:name])
+        if current_uuids.present?
+          module_by_name = current_cv_puppet_modules.find_by(:name => params[:name])
+          current_uuids.delete(module_by_name.uuid) if module_by_name && module_by_name.latest_in_modules_by_author?(query)
```

Comment:
Take example, there is one puppet module called "abc" having below versions:

name  version author
----------------------
abc   4.0.0   xyz
abc   3.0.0   xyz
abc   2.0.0   xyz
abc   1.0.2   pqr
abc   1.0.1   pqr
abc   1.0.0   pqr

With existing code,  User selected "4.0.0" version for content-view "test-cv". When he again clicked "select version", he will able to see below list:

Use Latest (currently 3.0.0)
abc   3.0.0   xyz
abc   2.0.0   xyz
Use Latest (currently 1.0.2)
abc   1.0.2   pqr
abc   1.0.1   pqr
abc   1.0.0   pqr

This "available_puppet_modules" API serves the results with & without name parameter for content-view.
"Use Latest" computation logic is totally present on client side in file [content-view.factory.js](engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/content-view.factory.js).

In this case, on the client side its  difficult to get latest version for selected puppet module as each version contains its own version value in computed version field and at this point don't have additional info about currently selected puppet module version.

Instead of skipping selected puppet module version which is latest module version among all versions of puppet module, I have added it in result list i.e by deleting the version from current_uuids list.

For verification whether this selected puppet module version is latest or not, I have added these additional queries.
By doing this, "Use latest" points to the latest version.

Does this make sense or should I change the result of this API on server side so that I get required information to client side for "Use Latest" computation?